snakemake/ioutils/input.py

```diff
@@ -0,0 +1,27 @@
+def parse_input(infile=None, parser=None, **kwargs):
+    def inner(wildcards, input, output):
+        with open(infile, "r") as fh:
+            if parser is None:
+                return fh.read().strip()
+            else:
+                return parser(fh, **kwargs)
+
+    return inner
+
+
+def extract_checksum(infile, **kwargs):
+    import pandas as pd
```

Comment:
Pandas is an optional dependency of snakemake. Hence we should have a try-catch here that raises a WorkflowError saying that pandas has to be installed in order to use the extract_checksum function.